File: 0088. Merge Sorted Array/0088.c
void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n) {
    // we're gonna try from the end to the beginning
    // nums1Size = m+n
    // nums2Size = n
    
    if(n==0) return;
    if(m==0) for(int i=0; i<n; i++) nums1[i]=nums2[i];
    m--;
    n--;
    while (m>=0 && n>=0){
        if(nums1[m]<nums2[n] || nums1[m]==nums2[n]) nums1[--nums1Size]=nums2[n--];
        else if(nums1[m]>nums2[n]) nums1[--nums1Size]=nums1[m--];
    }

    // if there remaining elements
    while(n>=0) nums1[--nums1Size]=nums2[n--];
}

File: 0088. Merge Sorted Array/0088.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        // size of nums1 is m+n but it has m elements. (other n elements are 0 and should be ignored.)
        // nums2 has n elements.
        // store them into nums1 as sorted.

        if(n==0){
            return;
        }
        int[] newArr = new int[m+n];
        int j=0; // pointer for nums1
        int k=0; // pointer for nums2
        for(int i=0;i<m+n;i++){
            System.out.print("i: " + i + " j: " + j + " k: " + k + "\n");
            if((k == n) || ((nums1[j]<nums2[k] || nums1[j]==nums2[k]) && j<m)){
                newArr[i]=nums1[j];
                j++;
            }
            else if(j==m || nums1[j]>nums2[k]){
                newArr[i]=nums2[k];
                k++;
            }
        }

        for(int i=0;i<m+n;i++){
            nums1[i]=newArr[i];
        }
    }
}